src/main/java/com/mongodb/jdbc/JsonSchema.java

```diff
@@ -0,0 +1,64 @@
+package com.mongodb.jdbc;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.bson.BsonInvalidOperationException;
+import org.bson.BsonString;
+import org.bson.BsonType;
+import org.bson.BsonValue;
+import org.bson.codecs.pojo.annotations.BsonIgnore;
+
+import static java.lang.String.format;
+
+// Simple POJO for deserializing jsonschema.
+public class JsonSchema {
+    private BsonValue polymorphicBsonType;
+    @BsonIgnore
+    public Set<String> bsonType;
```

Comment:
```suggestion
    public BsonValue bsonType;
```
I like the idea of having this `JsonSchema` class (for use as a deserialization target) as something separate from `MongoJsonSchema` (for using everywhere else in the API). I'd suggest that we go even further towards just making this class just a "dumb" deserialization target:
- keep the raw BsonValue for the `bsonType` field
- put whatever array/string polymophism-handling logic we need in the function that converts this to a `MongoJsonSchema`